adapter/analytics/analytics.go

```diff
@@ -236,16 +239,22 @@ func (m *manager) Close() {
 // uploadLoop runs a timer that periodically pushes everything in the buffer
 // directory to the server.
 func (m *manager) uploadLoop() {
```

Comment:
Can we see if the uploadLoop() can be abstracted to util package? That way we can test it once in util and reuse it in the various places we need it.